### PR TITLE
feat: return 0.0 when outside model bounds in TabulatedDynamics::computeContribution

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp [hunk 1]
```diff
@@ -130,6 +130,11 @@ VectorXd Tabulated::computeContribution(
         throw ostk::core::error::runtime::Wrong("Frame");
     }
 
+    if (anInstant < instants_.accessFirst() || anInstant > instants_.accessLast())
+    {
+        return VectorXd::Zero(interpolators_.getSize());
+    }
+
     const double epoch = (anInstant - instants_.accessFirst()).inSeconds();
```
Comment (apaletta3) [inline src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp:138]:
```suggestion
    const Real epoch = (anInstant - instants_.accessFirst()).inSeconds();
```

File: test/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.test.cpp [hunk 2]
```diff
@@ -214,15 +215,26 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Dynamics_Tabulated, Getters)
...
+            );
+        }
+
+        // Outside bounds
+        {
+            EXPECT_EQ(
+                tabulated.computeContribution(defaultInstants_[0] - Duration::Seconds(1.0), x, Frame::GCRF()),
+                VectorXd::Zero(3)
+            );
+        }
```
Comment (apaletta3) [inline test/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.test.cpp:237]:
```suggestion
        {
            EXPECT_EQ(
                tabulated.computeContribution(defaultInstants_[0] - Duration::Seconds(1.0), x, Frame::GCRF()),
                VectorXd::Zero(3)
            );
        }
        
        {
            EXPECT_EQ(
                tabulated.computeContribution(defaultInstants_[defaultInstants_.getSize() -1] + Duration::Seconds(1.0), x, Frame::GCRF()),
                VectorXd::Zero(3)
            );
        }
```

**Review by apaletta3 — APPROVED** [review]
Looks great!

Couple things:
- Can you remove the `[[maybe_unused]]` for the `aFrameSPtr` argument on line 124 since it is now actually used?
- Is there not already an out of data set guard in the interpolator? It might be worth adding at the interpolator level one so that our interpolation doesn't unwittingly turn into extrapolation

File: src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp (hunk 1 above)
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp:138]: The evaluator accepts a `double` https://github.com/open-space-collective/open-space-toolkit-mathematics/blob/93d7dc56e59d906c9c5527e828ce36f2f3a2225f/src/OpenSpaceToolkit/Mathematics/CurveFitting/Interpolator.cpp#L18 so it think it's better to be explicit.

**Comment by vishwa2710** [comment]
> Can you remove the [[maybe_unused]] for the aFrameSPtr argument on line 124 since it is now actually used?
Agreed!

> Is there not already an out of data set guard in the interpolator? It might be worth adding at the interpolator level one so that our interpolation doesn't unwittingly turn into extrapolation

There is! However it will throw an error. In this case we want to just return 0.0 instead of raising an error.

File: src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp (hunk 1 above)
Comment (apaletta3) [reply src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp:138]: Ok to keep it like this for now, but in the future perhaps better to align OSTk with using `Real` whenever possible?

**Review by apaletta3 — APPROVED** [review]
Dope. Good to resolve all changes and merge!

File: src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp (hunk 1 above)
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp:138]: Fair, I think we can update the interface there to accept a Real and then convert to a double when interfacing with boost.

**Pull request merged**